source/exceptions.tex

```diff
@@ -72,8 +72,11 @@
 \indextext{\idxcode{switch}!and try block}%
 \indextext{\idxcode{goto}!and handler}%
 \indextext{\idxcode{switch}!and handler}%
-A \tcode{goto} or \keyword{switch} statement shall not be used to transfer control
-into a try block or into a handler.
+A \tcode{goto} statement that transfers control to a statement within a try
+block or handler shall itself be within that try block or handler.
+A \tcode{case} or \tcode{default} label within a try block or handler shall be
+associated with a \tcode{switch} statement that is itself within that try block
+or handler.
```

Comment:
Hm... [stmt.if] p2 uses a slightly dfifferent phrasing:

> A case or default label appearing within such an if statement shall be associated with a switch statement
(8.5.3) within the same if statement. A label (8.2) declared in a substatement of a constexpr if statement
shall only be referred to by a statement (8.7.6) in the same substatement.

I think I like the cross-references. I also think we should express those restrictions with the same words, if we can. Bonus points for defining a term "limited control-flow statement" (or so) and just reusing that everywhere.

Reply:
I've added the new term to [stmt.label] and applied it to two places in [stmt.if] (in addition to the [except.pre] case).